server/mailet/mailets/src/main/java/org/apache/james/transport/util/SizeUtils.java

```diff
@@ -0,0 +1,34 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.transport.util;
+
+import com.jakewharton.byteunits.BinaryByteUnit;
+
+public class SizeUtils {
+
+    public static String humanReadableSize(long size) {
+        if (size < 0) {
+            return "-" + humanReadableSize(-size);
```

Comment:
IMO 

```
Precondition.checkAgument(size >= 0, "Size should be positive");
```

Reply:
We needed it before my refactoring

Reply:
anyay it can be useful, it is tested, I don't see why to not handle this case

Reply:
done but we do not the preconditions the underlying lib do it for us.

Reply:
@rouazana in fact if you have a negative size, this mean something is not done right or something when wrong. It's better to throw a exception then.
Moreover less code is better

Reply:
there are legitimate use of negative size, has it's a helper method, you should not make too much assumption about what is ok or not.